### PR TITLE
Add automatic publish to PyPI workflow with automatic release note

### PR DESCRIPTION
**Pull request by oncleben31**
The Release workflow publishes the package on PyPI. The workflow also creates a version tag in the GitHub repository, and publishes a GitHub Release using Release Drafter. The workflow is triggered on every push to the master branch.

Release steps only run if the package version was bumped. If the package version did not change, the package is instead uploaded to TestPyPI as a prerelease, and only a draft GitHub Release is created. TestPyPI is a test instance of the Python Package Index.

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+categories:
+  - title: ":boom: Breaking Changes"
+    label: "breaking"
+  - title: ":rocket: Features"
+    label: "enhancement"
+  - title: ":fire: Removals and Deprecations"
+    label: "removal"
+  - title: ":beetle: Fixes"
+    label: "bug"
```
Comment (Quentame) [inline .github/release-drafter.yml:9]:
```suggestion
    labels:
      - "bug"
      - "fix"
```

File: .github/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  - title: ":racehorse: Performance"
+    label: "performance"
+  - title: ":rotating_light: Testing"
+    label: "testing"
```
Comment (Quentame) [inline .github/release-drafter.yml:13]:
```suggestion
    labels:
      - "test"
      - "testing"
```

File: .github/release-drafter.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  - title: ":construction_worker: Continuous Integration"
+    label: "ci"
+  - title: ":books: Documentation"
+    label: "documentation"
```
Comment (Quentame) [inline .github/release-drafter.yml:17]:
```suggestion
    labels:
      - "doc"
      - "documentation"
```

File: .github/release-drafter.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  - title: ":hammer: Refactoring"
+    label: "refactoring"
+  - title: ":lipstick: Style"
+    label: "style"
+  - title: ":package: Dependencies"
+    labels:
+      - "dependencies"
```
Comment (Quentame) [inline .github/release-drafter.yml:24]:
```suggestion
      - "deps"
      - "dependency"
      - "dependencies"
```

File: .github/release-drafter.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  - title: ":racehorse: Performance"
+    label: "performance"
+  - title: ":rotating_light: Testing"
+    label: "testing"
+  - title: ":construction_worker: Continuous Integration"
+    label: "ci"
+  - title: ":books: Documentation"
+    label: "documentation"
+  - title: ":hammer: Refactoring"
+    label: "refactoring"
```
Comment (Quentame) [inline .github/release-drafter.yml:19]:
```suggestion
    labels:
      - "refactore"
      - "refactoring"
```

**Review by Quentame — APPROVED** [review]
Wow, very cool CI/CD :+1: :tada: 

Seems nice!

Added some label release propositions.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Detect and tag new version
+        id: check-version
+        if: steps.check-parent-commit.outputs.sha
+        uses: salsify/action-detect-and-tag-new-version@v2.0.1
+        with:
+          version-command: |
+            bash -o pipefail -c "poetry version | awk '{ print \$2 }'"
+
+      - name: Bump version for developmental release
```
Comment (Quentame) [inline .github/workflows/release.yml:46]:
```suggestion
      - name: Bump version for development release
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Publish package on TestPyPI
+        if: "! steps.check-version.outputs.tag"
+        uses: pypa/gh-action-pypi-publish@v1.4.1
+        with:
+          user: __token__
+          password: ${{ secrets.TEST_PYPI_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
+
+      - name: Publish the release notes
```
Comment (Quentame) [inline .github/workflows/release.yml:72]:
```suggestion
      - name: Publish release notes
```

**Review by Quentame — CHANGES_REQUESTED** [review]
Ups, some English corrections

File: .github/release-drafter.yml (hunk 1 above)
Comment (oncleben31) [reply .github/release-drafter.yml:9]: Are you sure this is necessary? We should avoid similar labels to faciltate issue and PR triage. As admin we are choosing the label. I don't understand the need to have different wording for the same label type.

For information the labels have been already created in this repository.

Comment (Quentame) [reply .github/release-drafter.yml:9]: Oh yes good !
I thought it was the text included in the PR title :sweat_smile: 

So it should have one everywhere sure :+1:

**Review by Quentame — APPROVED** [review]

**Comment by oncleben31** [comment]
Contributes to #18

**Pull request merged**